### PR TITLE
SUPESC-144: Added integration of `CartItemPricePreCheckPlugin` plugin

### PR DESCRIPTION
**Pull request by Zingeon**
## PR Description

Ticket: https://spryker.atlassian.net/browse/SUPESC-144

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 1]
```diff
@@ -0,0 +1,869 @@
...
+This article describes how to add product variants and product images to an existing cart.
+{% endinfo_block %}
+
+## Prerequisites:
+
+Before starting make sure you are familiar with the concept of Spryker Super Attributes.
+
+## UI Changes:
+
+Cart now supports changing the items in the cart by modifying their attributes. If we have a wrong T-Shirt size in the cart we will be able to change it.
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:26]:
```suggestion
Cart now supports changing cart items by modifying their attributes. If we have a wrong T-Shirt size in the cart we will be able to change it.
```

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 2]
```diff
@@ -0,0 +1,869 @@
...
+
+Cart now also supports product images out of the box.
+![cart_product_images](https://spryker.s3.eu-central-1.amazonaws.com/docs/scos/dev/feature-integration-guides/cart-integration.md/cart_product_images.png)
+
+If we have products with multiple super attributes we can now, narrowing-down in the cart.
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:31]: we can now do what?

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 3]
```diff
@@ -0,0 +1,869 @@
...
+protected function getExpanderPlugins(Container $container)
+{
+    return [
+        // your existing plugins ...
+        new ProductImageCartPlugin(),
+    ];
+}
+```
+
+If your shop uses product bundles, register `ExpondBundleItemsWithImagesPlugin` in your shop's `CartDependencyProvider` as follows:
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:61]:
```suggestion
If your shop uses product bundles, register `ExpandBundleItemsWithImagesPlugin` in your shop's `CartDependencyProvider` as follows:
```

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 4]
```diff
@@ -0,0 +1,869 @@
...
+use Generated\Shared\Transfer\StorageProductTransfer;
+use Pyz\Yves\Cart\Handler\CartItemHandlerInterface;
+use Spryker\Yves\CartVariant\Dependency\Plugin\CartVariantAttributeMapperPluginInterface;
+class AttributeVariantsProvider
+{
+    /**
+     * @var \Spryker\Yves\CartVariant\Dependency\Plugin\CartVariantAttributeMapperPluginInterface
+     */
+    protected $cartVariantAttributeMapperPlugin;
+    /**
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:246]: new line required

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 5]
```diff
@@ -0,0 +1,869 @@
...
+     * @var \Pyz\Yves\Cart\Handler\CartItemHandlerInterface
+     */
+    protected $cartItemHandler;
+    /**
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:250]: same

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 6]
```diff
@@ -0,0 +1,869 @@
...
+     * @param \Pyz\Yves\Cart\Handler\CartItemHandlerInterface $cartItemHandler
+     */
+    public function __construct(
+        CartVariantAttributeMapperPluginInterface $cartVariantAttributeMapperPlugin,
+        CartItemHandlerInterface $cartItemHandler
+    ) {
+        $this->cartVariantAttributeMapperPlugin = $cartVariantAttributeMapperPlugin;
+        $this->cartItemHandler = $cartItemHandler;
+    }
+    /**
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:263]: Please check all cases

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 7]
```diff
@@ -0,0 +1,869 @@
...
+use Spryker\Client\Product\ProductClientInterface;
+use Spryker\Shared\CartVariant\CartVariantConstants;
+use Spryker\Yves\Messenger\FlashMessenger\FlashMessengerInterface;
+class CartItemHandler extends BaseHandler implements CartItemHandlerInterface
+{
+    /**
+     * @var \Pyz\Yves\Cart\Handler\CartOperationInterface
+     */
+    protected $cartOperationHandler;
+    /**
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:351]: same here

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 8]
```diff
@@ -0,0 +1,869 @@
...
+     * @return array
+     */
+    protected function getAvailableAttributesForItem(ArrayObject $items, array $itemAttributes, $sku)
+    {
+        $storageProductTransfer = $this->getProductStorageTransfer($sku, $itemAttributes[$sku], $items);
+        $availableAttributes = $storageProductTransfer->getAvailableAttributes();
+        return $availableAttributes;
+    }
+    /**
+     * @SuppressWarnings(PHPMD.UnusedLocalVariable)
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:511]: IMO this should not be on the doc

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 9]
```diff
@@ -0,0 +1,869 @@
...
+```php
+<?php
+/**
+ * This file is part of the Spryker Demoshop.
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+namespace Pyz\Yves\Cart\Handler;
+use ArrayObject;
+use Generated\Shared\Transfer\StorageProductTransfer;
+interface CartItemHandlerInterface
```
Comment (kkicheglovspryker) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:550]: same, no line breaks

**Review by kkicheglovspryker — CHANGES_REQUESTED** [review]

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md (hunk 2 above)
Comment (dmytro-sokolovskyi) [reply docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:31]: Haven't fixed.

Comment (andriitserkovnyi) [reply docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:31]: Based on the super attributes, you can select the needed product variant in cart.

File: docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+redirect_from:
+  - /2021080/docs/cart-integration
+  - /2021080/docs/en/cart-integration
+  - /docs/cart-integration
+  - /docs/en/cart-integration
+---
+
+{% info_block infoBox %}
+
+This article describes how to add product variants and product images to an existing cart.
```
Comment (dmytro-sokolovskyi) [inline docs/scos/dev/feature-integration-guides/upcoming-release/cart-integration.md:17]: We should also mention a price validation here.

**Review by dmytro-sokolovskyi — DISMISSED** [review]

File: docs/scos/dev/feature-integration-guides/202212.0/cart-integration.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+
+{% info_block infoBox %}
+
+This document describes how to add product variants, product images, and price validation for a cart's items to an existing cart.
+
+{% endinfo_block %}
+
+## Prerequisites
+
+Read the concept of Spryker [super attributes](/docs/scos/user/features/{{page.versionj}}/product-feature-overview/product-attributes-overview.html#super-attributes).
```
Comment (VadymSachenko) [inline docs/scos/dev/feature-integration-guides/202212.0/cart-integration.md:23]:
```suggestion
Read the concept of Spryker [super attributes](/docs/scos/user/features/{{page.version}}/product-feature-overview/product-attributes-overview.html#super-attributes).
```

**Review by dmytro-sokolovskyi — APPROVED** [review]

**Review by VadymSachenko — DISMISSED** [review]

**Review by VadymSachenko — APPROVED** [review]

**Review by kkicheglovspryker — APPROVED** [review]

**Pull request merged**